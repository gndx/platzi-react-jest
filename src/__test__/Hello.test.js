import React from 'react';
import Hello from '../components/Hello';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('AÃ±adiendo pruebas a React', () => {
  test('Comprobamos el texto que recibe como prop', () => {
    const data = { title: 'Hola Jest' };
    const wrapper = mount(
      <Hello title={data.title} />
    );
    const p = wrapper.find('h1');
    expect(p.text()).toBe('Hola Jest');
  });

  test('Creamos un snapshot en React', () => {
      const data = { title: 'Hola Jest' };
      const wrapper = shallow(
        <Hello title={data.title} />
      );
      expect(wrapper).toMatchSnapshot();
  });

});

