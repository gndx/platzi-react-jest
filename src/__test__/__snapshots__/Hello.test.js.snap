// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AÃ±adiendo pruebas a React Creamos un snapshot en React 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hello
    title="Hola Jest"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hola Jest",
    },
    "ref": null,
    "rendered": "Hola Jest",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hola Jest",
      },
      "ref": null,
      "rendered": "Hola Jest",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
